scripts/check-scripts.ts

```diff
@@ -53,10 +55,70 @@ const perPackageExceptions: Record<string, Record<string, () => string | undefin
 	},
 }
 
+const packagesWithoutTSConfigs: ReadonlySet<string> = new Set(['config'])
+
+async function checkTsConfigs({ packages, fix }: { fix?: boolean; packages: Package[] }) {
+	for (const workspace of packages) {
+		const tsconfigPath = path.join(workspace.path, 'tsconfig.json')
+		if (packagesWithoutTSConfigs.has(workspace.name)) {
+			continue
+		}
+
+		if (!existsSync(tsconfigPath)) {
+			throw new Error('No tsconfig.json found at ' + tsconfigPath)
+		}
+
+		const tsconfig = parse(readFileSync(tsconfigPath, 'utf-8'))
+		const tldrawDeps = Object.keys({
+			...workspace.packageJson.dependencies,
+			...workspace.packageJson.devDependencies,
+		}).filter((dep) => dep.startsWith('@tldraw/'))
+
+		const fixedDeps = tsconfig.references ?? []
+		const missingRefs = []
+		for (const dep of tldrawDeps) {
+			// construct the expected path to the dependency's tsconfig
+			const matchingWorkspace = packages.find((p) => p.name === dep)
+			if (!matchingWorkspace) {
+				throw new Error(`No workspace found for ${dep}`)
+			}
+			const tsconfigReferencePath = path.relative(workspace.path, matchingWorkspace.path)
+			if (
+				!tsconfig.references?.some(({ path }: { path: string }) => path === tsconfigReferencePath)
+			) {
+				fixedDeps.push({ path: tsconfigReferencePath })
+				missingRefs.push(dep)
+			}
+		}
+		if (missingRefs.length) {
+			if (fix) {
+				tsconfig.references = fixedDeps.sort((a: any, b: any) => a.path.localeCompare(b.path))
+				writeFileSync(tsconfigPath, JSON.stringify(tsconfig, null, '\t'), 'utf-8')
```

Comment:
Using our `writeJsonFile` helper here will make sure that the generated files match our prettier formatting